client/src/main/scala/drt/client/components/FlightTable.scala

```diff
@@ -96,6 +107,8 @@ object FlightTable {
           ^.autoFocus := true,
           ^.onChange ==> { e: ReactEventFromInput =>
             val value = e.target.value
+            typingSearchTimer = Some(setTimeout(doneSearchTypingInterval)(doneSearchTyping(latestSearchTypingValue, props.airportConfig.portCode.toString)))
```

Comment:
Looks like you're passing the previous value to the timer callback rather than the new value. You can pass `e.target.value` straight to `doneSearchTyping` here and get rid of `latestSearchTypingValue`